swarm_copy_tests/app/routers/test_threads.py

```diff
@@ -0,0 +1,125 @@
+from unittest.mock import AsyncMock, Mock, patch
+
+import pytest
+
+from swarm_copy.app.database.sql_schemas import utc_now
+
+
+@pytest.mark.asyncio
+async def test_create_thread(app_client, settings):
+    mock_validate_project = AsyncMock()
+    mock_session = AsyncMock()
+
+    def add_thread(thread):
+        thread.creation_date = utc_now()
+        thread.update_date = utc_now()
+        thread.thread_id = "thread_id"
+
+    mock_session.add = add_thread
+    user_id = "user_id"
+    token = "token"
+    title = "title"
+    project_id = "project_id"
+    virtual_lab_id = "virtual_lab_id"
+    with patch("swarm_copy.app.app_utils.validate_project", mock_validate_project):
+        from swarm_copy.app.routers.threads import create_thread
+        await create_thread(app_client, settings,
+                            token,
+                            virtual_lab_id,
+                            project_id,
+                            mock_session,
+                            user_id,
+                            title)
+
+    assert mock_session.commit.called
+    assert mock_session.refresh.called
```

Comment:
This is not really testing the endpoint. I suggest we use the real databases we have available and use the dependency overwrite logic of FastAPI as we did in the equivalent test in `neuroagent`.

Reply:
Ok, I changed the test and all the others to use endpoints.